### ISSUE TITLE
Incident report, which may become repo2docker bug report.

### ISSUE THREAD

**Issue by consideRatio** [neurohackademy/nh2020-jupyterhub#37]
There was an issue with #36 that wasn't apparent, which made the startup of pods fail because the `jupyterhub-singleuser` executable script which comes by installing the jupyterhub package wasn't around any more.

I've debugged it and worked around it for now by installing jupyterhub with conda, but I still don't understand why `conda update jupyterlab` in the postBuild script made 24 entries in `/srv/conda/envs/notebook/bin/` disappear, one of them was `jupyterhub-singleuser`.

Apparently, the combination of having environment.yml (not happening with requirements.txt), and having postBuild running `conda update jupyterlab` is what triggers this. Below are the executables that disappeared from `/srv/conda/envs/notebook/bin/`.

```
{'alembic',
 'certipy',
 'compile_et',
 'curl-config',
 'gss-client',
 'jupyter-labhub',
 'jupyter-nteract',
 'jupyterhub',
 'jupyterhub-singleuser',
 'k5srvutil',
 'kadmin',
 'kdestroy',
 'kinit',
 'klist',
 'kpasswd',
 'krb5-config',
 'ksu',
 'kswitch',
 'ktutil',
 'kvno',
 'mako-render',
 'pyjwt',
 'sclient',
 'sim_client',
 'uuclient'}
```

**Comment by consideRatio** [comment]
`conda update jupyterlab` made the following happen.

```
The following packages will be REMOVED:

  alembic-1.3.3-py_0
  async_generator-1.10-py_0
  blinker-1.4-py_1
  certipy-0.1.3-py_0
  inflect-4.0.0-py37_1
  ipywidgets-7.5.1-py_0
  jaraco.itertools-5.0.0-py_0
  jupyter_telemetry-0.0.4-py_0
  jupyterhub-base-1.1.0-py37_2
  jupyterhub-singleuser-1.1.0-py37_2
  krb5-1.16.4-h2fd8d38_0
  libcurl-7.65.3-hda55be3_0
  libedit-3.1.20170329-hf8c457e_1001
  libssh2-1.8.2-h22169c7_2
  mako-1.1.0-py_0
  more-itertools-8.2.0-py_0
  nteract_on_jupyter-2.1.3-py_0
  oauthlib-3.0.1-py_0
  pamela-1.0.0-py_0
  prompt_toolkit-3.0.3-py_0
  pycurl-7.43.0.5-py37h16ce93b_0
  pyjwt-1.7.1-py_0
  python-editor-1.0.4-py_0
  python-json-logger-0.1.11-py_0
  ruamel.yaml-0.16.6-py37h516909a_0
  ruamel.yaml.clib-0.2.0-py37h516909a_0
  sqlalchemy-1.3.13-py37h516909a_0
  widgetsnbextension-3.5.1-py37_0
```

**Comment by consideRatio** [comment]
Since these issues came from `conda update jupyterlab` without having `jupyterhub` explicitly installed I became afraid of using it entirely, as its not only JupyterHub that goes away when this happen but for example ipywidgets available by default in repo2docker images. Due to that, I removed that part from postBuild which we had to update JupyterLab.

To still use JupyterLab 2 I learned that we can add it explicitly and unpinned. And like that resolve this issue.
